journalpump/journalpump.py

```diff
@@ -495,6 +496,14 @@ def value_func(tags, f=f, args=args):  # pylint: disable=undefined-variable
 
             yield output
 
+    def _build_secret_filters(self, config):
+        secret_filters = config.get("secret_filters")
+        if secret_filters is not None:
+            for idx, _ in enumerate(secret_filters):
+                secret_filters[idx]["compiled_pattern"] = re.compile(secret_filters[idx]["pattern"])
```

Comment:
This is optimistic dict access, with an invalid config raising an exception. Need to either validate the `secret_filters` entry has a valid structure, use safe access, or a `try`/`except KeyError` block to handle the possibility.

Reply:
Need to also protect against an invalid pattern being provided when compiling.

Reply:
What would the ideal behavior be in the case of an invalid regex? 

I think I should add a try catch, but mostly for a neater looking error message. An invalid regex in the config should probably still exit journalpump. The alternative would be dropping the config and running in a degraded state (without filtering) but I think that's a bad pattern as it lets the issue go unnoticed.

Reply:
Hmm. I think mostly with a resiliency cap on - I'd rather run in a degraded state versus not running at all, but can see the benefits of simply crashing. I'll +1 whatever you decide on this, but at least giving a nice error message would be good :)

Reply:
I'm going to wear my security cap here and lean towards saying it would be better to send nothing at all, than to think you are protected by filtering which was disabled. In the case of something like an API service, I'd probably rather have partly working than completely broken.